File: apiEZPZ/src/main/java/com/example/apiezpz/search/repository/WeeklyRankRepository.java
package com.example.apiezpz.search.repository;

import com.example.apiezpz.search.entity.WeeklyRank;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.time.LocalDate;
import java.util.List;

@Repository
public interface WeeklyRankRepository extends JpaRepository<WeeklyRank, Long> {
    List<WeeklyRank> findByRecordedDate(LocalDate date);

    List<WeeklyRank> findByStartDateBetween(LocalDate startDate, LocalDate endDate);


}

File: apiEZPZ/src/main/kotlin/com/example/apiezpz/search/repository/WeeklyRankRepository.kt
package com.example.apiezpz.search.repository

import com.example.apiezpz.search.entity.WeeklyRank
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.LocalDate

@Repository
interface WeeklyRankRepository : JpaRepository<WeeklyRank, Long> {
    fun findByRecordedDate(date: LocalDate): List<WeeklyRank>
    fun findByStartDateBetween(startDate: LocalDate, endDate: LocalDate): List<WeeklyRank>
    fun findTop10ByOrderBySearchCountDesc(): List<WeeklyRank>
} 